src/Integration.UnitTests/Binding/ProjectSystemFilterTests.cs

```diff
@@ -0,0 +1,171 @@
+        private ConfigurableVsProjectSystemHelper projectSystem;
+
+        [TestInitialize]
+        public void TestInitialize()
+        {
+            this.serviceProvider = new ConfigurableServiceProvider();
+            this.projectSystem = new ConfigurableVsProjectSystemHelper(this.serviceProvider);
+            this.serviceProvider.RegisterService(typeof(IProjectSystemHelper), this.projectSystem);
+        }
+
+        #region Tests 
+
+        [TestMethod]
+        public void ProjectSystemFilter_IsAccepted_UnsupportedLanguage_IsFalse()
+        {
+            // Setup
+            var testSubject = this.CreateTestSubject();
+
+            var project = new ProjectMock("unsupported.vbproj");
+            project.SetVBProjectKind();
+
+            // Act
+            var result = testSubject.IsAccepted(project);
+
+            // Verify
+            Assert.IsFalse(result, "Project of unsupported language type should not be accepted");
+        }
+
+        [TestMethod]
+        public void ProjectSystemFilter_IsAccepted_SonarExcludeProperty_ReturnsPropertyValue()
+        {
+            // Setup
+            var testSubject = this.CreateTestSubject();
+
+            var project = new ProjectMock("supported.csproj");
+            project.SetCSProjectKind();
+
+            // Test case 1: property true -> not accepted
+            // Setup
+            project.SetBuildProperty(Constants.SonarQubeExcludeBuildPropertyKey, "true");
+            
+            // Act
+            var result = testSubject.IsAccepted(project);
+
+            // Verify
+            Assert.IsFalse(result, "Project with SonarExcludeProperty=false should NOT be accepted");
+
+            // Test case 2: property false -> is accepted
+            // Setup
+            project.SetBuildProperty(Constants.SonarQubeExcludeBuildPropertyKey, "false");
+
+            // Act
+            result = testSubject.IsAccepted(project);
+
+            // Verify
+            Assert.IsTrue(result, "Project with SonarExcludeProperty=true should be accepted");
+        }
+
+        [TestMethod]
+        public void ProjectSystemFilter_IsAccepted_SonarTestProjectProperty_ReturnsPropertyValue()
+        {
+            // Setup
+            var testSubject = this.CreateTestSubject();
+
+            var project = new ProjectMock("supported.csproj");
+            project.SetCSProjectKind();
+
+            // Test case 1: property true -> not accepted
+            // Setup
+            project.SetBuildProperty(Constants.SonarQubeTestProjectBuildPropertyKey, "true");
+
+            // Act
+            var result = testSubject.IsAccepted(project);
+
+            // Verify
+            Assert.IsFalse(result, "Project with SonarQubeTestProject=false should NOT be accepted");
+
+            // Test case 2: property false -> is accepted
+            // Setup
+            project.SetBuildProperty(Constants.SonarQubeTestProjectBuildPropertyKey, "false");
```

Comment:
Same question as above (what if the value is empty, or in more general terms cannot be parsed to bool)?